crypto/evp/p_lib.c

```diff
@@ -106,43 +120,101 @@ int EVP_PKEY_copy_parameters(EVP_PKEY *to, const EVP_PKEY *from)
         return 0;
     }
 
-    if (from->ameth && from->ameth->param_copy)
+    /*
+     * Only use the routine for provided keys if either key is purely provided.
```

Comment:
This comment also doesn't match the logic below (which is both keys)

Reply:
I need #11074 in place before tackling this one